### PR TITLE
Add OwLab Spring files

### PR DESCRIPTION
**Pull request by owlab-git**
## Description

Added support for the PCB of the OwLab Spring.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/owlab/spring/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		         KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS
+	),
+
+	[3] = LAYOUT_spring(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		         KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS
+	)
```
Comment (drashna) [inline keyboards/owlab/spring/keymaps/default/keymap.c:52]:
```suggestion
```

File: keyboards/owlab/spring/spring.c [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "spring.h"
+#include "rgblight.h"
```
Comment (drashna) [inline keyboards/owlab/spring/spring.c:18]:
```suggestion
```

File: keyboards/owlab/spring/spring.c [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+enum caps_modes{
+    CAPS_MODE_UPPER = 0, //UPPER CASE 
+    CAPS_MODE_LOWER      //LOWER CASE
+};
+
+uint8_t caps_mode_index;
+uint8_t pre_hue, pre_sat, pre_val; 
+uint8_t previous_rgb_mode = 0;
```
Comment (drashna) [inline keyboards/owlab/spring/spring.c:27]: It may be simpler to just do:
```suggestion
rgblight_config_t prev_config;
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by owlab-git** [comment]
Hi, We had modified as suggestion, Please check, Thanks.

File: keyboards/owlab/spring/config.h [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* NKRO */
+#define FORCE_NKRO 
+
+/* RGB Strip*/
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/owlab/spring/config.h:49]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/owlab/spring/config.h (hunk 4 above)
Comment (owlab-git) [reply keyboards/owlab/spring/config.h]: Yes. Thanks for the suggestion.

**Review by drashna — APPROVED** [review]

File: keyboards/owlab/spring/rules.mk [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+BOOTLOADER = atmel-dfu
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes  # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes   # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = no     # Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+NKRO_ENABLE = yes       # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/owlab/spring/rules.mk:13]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes  # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes   # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = no     # Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
NKRO_ENABLE = yes       # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by owlab-git** [comment]
Hi, rules.mk has been modified as suggestion. Please check again, thanks.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**